presto-geospatial/src/test/java/com/facebook/presto/geospatial/TestGeoQueries.java

```diff
@@ -0,0 +1,346 @@
+        assertQuery("select ST_NumPoints(ST_GeometryFromText('LineString(8 4, 5 7)'))", "select 2");
+        assertQuery("select ST_NumPoints(ST_GeometryFromText('Point (1 2)'))", "select 1");
+        assertQuery("select ST_NumPoints(ST_GeometryFromText('LineString EMPTY'))", "select 0");
+    }
+
+    @Test
+    public void testSTIsRing()
+            throws Exception
+    {
+        assertQuery("select ST_IsRing(ST_GeometryFromText('LineString(8 4, 4 8)'))", "select false");
+        assertQueryFails("select ST_IsRing(ST_GeometryFromText('Polygon ((2 0, 2 1, 3 1))'))", "ST_IsRing only applies to LineString. Input type is: Polygon");
+    }
+
+    @Test
+    public void testSTStartEndPoint()
+            throws Exception
+    {
+        assertQuery("select ST_AsText(ST_StartPoint(ST_GeometryFromText('LineString(8 4, 4 8, 5 6)')))", "select 'POINT (8 4)'");
+        assertQuery("select ST_AsText(ST_EndPoint(ST_GeometryFromText('LineString(8 4, 4 8, 5 6)')))", "select 'POINT (5 6)'");
+        assertQueryFails("select ST_AsText(ST_EndPoint(ST_GeometryFromText('Polygon ((2 0, 2 1, 3 1))')))", "ST_EndPoint only applies to LineString. Input type is: Polygon");
+    }
+
+    @Test
+    public void testSTXY()
+            throws Exception
+    {
+        assertQuery("select ST_X(ST_GeometryFromText('Point (1 2)'))", "select 1.0");
+        assertQuery("select ST_Y(ST_GeometryFromText('Point (1 2)'))", "select 2.0");
+        assertQueryFails("select ST_Y(ST_GeometryFromText('Polygon ((2 0, 2 1, 3 1))'))", "ST_Y only applies to Point. Input type is: Polygon");
+    }
+
+    @Test
+    public void testSTBoundary()
+            throws Exception
+    {
+        assertQuery("select ST_AsText(ST_Boundary(ST_GeometryFromText('Polygon ((1 1, 4 1, 1 4))')))", "select 'LINESTRING (1 1, 4 1, 1 4, 1 1)'");
+    }
+
+    @Test
+    public void testSTEnvelope()
+            throws Exception
+    {
+        assertQuery("select ST_AsText(ST_Envelope(ST_GeometryFromText('LineString(1 1, 2 2, 1 3)')))", "select 'POLYGON ((1 1, 2 1, 2 3, 1 3, 1 1))'");
+    }
+
+    @Test
+    public void testSTDifference()
+            throws Exception
+    {
+        assertQuery("select ST_AsText(ST_Difference(ST_GeometryFromText('Polygon ((1 1, 1 4, 4 4, 4 1))'), ST_GeometryFromText('Polygon ((2 2, 2 5, 5 5, 5 2))')))", "select 'POLYGON ((1 1, 4 1, 4 2, 2 2, 2 4, 1 4, 1 1))'");
+    }
+
+    @Test
+    public void testSTDistance()
+            throws Exception
+    {
+        assertQuery("select ST_Distance(ST_GeometryFromText('Polygon ((1 1, 1 3, 3 3, 3 1))'), ST_GeometryFromText('Polygon ((4 4, 4 5, 5 5, 5 4))'))", "select 1.4142135623730951");
+    }
+
+    @Test
+    public void testSTExteriorRing()
+            throws Exception
+    {
+        assertQuery("select ST_AsText(ST_ExteriorRing(ST_GeometryFromText('Polygon ((1 1, 1 4, 4 1))')))", "select 'LINESTRING (1 1, 4 1, 1 4, 1 1)'");
```

Comment:
nit: would you add a test for an empty polygon and a polygon with one or more holes?